### PR TITLE
Replace stats client with baseplate metrics client.

### PR DESCRIPTION
**Pull request by dellis23**

### CODE REVIEW

**Comment by dellis23** [comment]
:eyeglasses: @spladug 

Need to merge puppet changes before this.

File: events/injector.py [hunk 1]
```diff
@@ -171,9 +173,9 @@ def main():
     region = config["aws.region"]
     kinesis = boto.kinesis.connect_to_region(region)
 
-    stats_client = events.stats.make_stats_client(config)
+    metrics_client = base_plate.make_metrics_client(config)
```
Comment (spladug) [inline events/injector.py]: base_plate ?

File: tests/collector_tests.py [hunk 2]
```diff
@@ -51,11 +51,11 @@ def utcnow(cls):
         }
         self.event_sink = MockSink()
         self.error_sink = MockSink()
-        self.mock_stats_client = mock.MagicMock()
+        self.mock_metrics_client = mock.MagicMock()
```
Comment (spladug) [inline tests/collector_tests.py]: I was dumb when I wrote these tests, worth adding autospec to the mock?

**Comment by spladug** [comment]
:nail_care: just a couple little things. looks nice! exciting to see the duplicate code drop away :)

File: events/injector.py (hunk 1 above)
Comment (dellis23) [inline events/injector.py]: :facepalm:

**Comment by dellis23** [comment]
:haircut:

**Comment by spladug** [comment]
:fish: looks great!

**Pull request merged**